libs/utils/src/seqwait.rs

```diff
@@ -43,9 +43,23 @@ where
 {
     waiters: BinaryHeap<Waiter<V>>,
     current: S,
+    status_channel: Sender<Option<V>>,
     shutdown: bool,
 }
 
+impl<S, V> SeqWaitInt<S, V>
+where
+    S: MonotonicCounter<V>,
+    V: Ord + Copy,
+{
+    /// `status_channel` contains the number of the first waiter in the queue.
+    /// This function should be called whenever waiters queue changes.
+    fn update_status(&self) {
```

Comment:
I would have preferred a more robust approach than making devs remember to call this function whenever the `self.waiters` changes .

For example, we could have wrapped the `BinaryHeap<Waiter<V>>` with a type whose job it is to update the status.
Like

```rust
struct Waiters<V> {
  heap: BinaryHeap<Waiter<V>>,
  status: watch::Sender<Option<V>>,
}

impl Waiters {
  pub fn push(...) {
    self.heap.push(...);
    self.status.send_replace(...);
  }
}

impl Drop for Waiters {
  fn drop(...) {
    self.status.send_replace(None);
  }
}


Instead of putting `status_channel` aside `waiters`, I